### ISSUE TITLE
StreamSubscription.cancel() not working

### ISSUE THREAD

**Issue by arnold-parge** [zino-app/graphql-flutter#126]
<hr>

<h1>StreamSubscription.cancel() not working</h1>
Something like `{type: unknown, payload: {type: ka}}` keeps getting printed in my console even after cancelling the StreamSubscription.

<hr>

**To Reproduce**
Steps to reproduce the behaviour:
1. Start Subscription  `{type: unknown, payload: {type: ka}}` keeps getting printed in console
2. Put StreamSubscription to global variable
3. Cancel Subscription
4. `{type: unknown, payload: {type: ka}}` still keeps getting printed in console

<hr>

**Expected behaviour**
Subscription should stop on `cancel()` 

<hr>

**System Details**
 - OS: Ubuntu
 - Version: 17.10

<hr>

**Device Details:**
 - Device: Mi A1
 - OS: Android Oreo
 - Version: 8.1.0

<hr>

**Complete Code**
Data is fetched perfectly, but **I don't want unwanted connections remaining** with my server, when not needed (even after cancelling the subscription).
Below is the complete code to refer:
```
class LveScreenState extends State<LveScreen> {
  String operationName = 'operationName';
  StreamSubscription streamSubscription;

  List patients = [];

  String query = '';

  void initSocket() async {
    query = """
      subscription $operationName {
        patient(limit:2, order_by:id_desc) {
          name
          email
        }
      }
    """
        .replaceAll('\n', ' ');

    SocketClient socketClient =
        await SocketClient.connect('ws://some.ip.address:8080/v1alpha1/graphql');
    streamSubscription = socketClient
        .subscribe(SubscriptionRequest(operationName, query, {}))
        .listen((subData) {
      setState(() {
        this.patients = subData.toJson()['data']['patient'];
        print('patients: $patients');
      });
    });
  }

  @override
  void initState() {
    super.initState();
    this.initSocket();
  }

  Future<bool> cancelSubs() async {
    await streamSubscription.cancel();
    return true;
  }

  @override
  Widget build(BuildContext context) {
    return WillPopScope(
      child: new Scaffold(
        appBar: new AppBar(
          title: new Text('Homepage'),
        ),
        body: ListView.builder(
          itemBuilder: (context, i) {
            return ListTile(
              title: Text(patients[i]['name']),
              subtitle: Text(patients[i]['email']),
            );
          },
          itemCount: patients.length,
        ),
      ),
      onWillPop: () => cancelSubs(),
    );
  }
}

```

**Comment by ArneSchulze** [comment]
Canceling the `StreamSubscription` object you received from calling `socketClient.subscribe(...).listen(...)` actually just cancels the data stream subscription you provided in `listen()`. To terminate the WebSocket connection to your server you would have to send a `GQL_CONNECTION_TERMINATE` message, so your GraphQL-Implementation on your server knows that you won't continue querying. After that, you would have to terminate the WebSocket connection itself.

**Comment by ArneSchulze** [comment]
Unfortunately I haven't found a solution yet to cancel the WebSocket with the current `SocketClient` implementation. Maybe I will create a PullRequest regarding this issue.

**Comment by HofmannZ** [comment]
Fixed with #147
